modules/NetworkSplitter3D/src/main/java/org/alexandrebarao/networksplitter3d/NetworkSplitter3D.java

```diff
@@ -0,0 +1,502 @@
+                    "Layer distance between each Z-Level layout. Used to multiply Z-Scale. Note: you can use the RESET button to layout your original network.",
+                    "getZdistance", "setZdistance"));
+            
+             properties.add(LayoutProperty.createProperty(
+                    this, Float.class,
+                    "Z-Scale",
+                    NETWORKSPLITTER3D,
+                    "The vertical Z-Scale (pixels).  Note: you can use the RESET button to layout your original network.",
+                    "getVerticalScale", "setVerticalScale"));
+                     
+              properties.add(LayoutProperty.createProperty(
+                    this, Double.class,
+                    "Alfa",
+                    NETWORKSPLITTER3D,
+                    "The rotation angle (degrees from 0 to 360). Note: you can use the RESET button to layout your original network.",
+                    "getAlfa", "setAlfa"));
+              
+                         
+            
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return properties.toArray(new LayoutProperty[0]);
+    }
+
+     public double getDataLaboratoryValue(Node n, Column col) {
+        
+                   double z = 0f;
+                   
+                   
+                   if ( col.getTypeClass()== String.class ) { 
+                        String colString = (String) (n.getAttribute(col));
+                        z = Double.valueOf(colString).doubleValue();
+                   }
+                   
+                   if ( col.getTypeClass()== Integer.class ) { 
+                        Integer colString = (Integer) (n.getAttribute(col));
+                        z = colString.intValue();
+                   }
+                   
+                   if ( col.getTypeClass()== Float.class ) { 
+                        Float colString = (Float) (n.getAttribute(col));
+                        z = colString.floatValue();
+                   }
+                   
+                   if ( col.getTypeClass()== Double.class ) { 
+                        Double colString = (Double) (n.getAttribute(col));
+                        z = colString.doubleValue();
+                   }
+                   
+                   return z;
+    }
+    
+     
+      public Node searchNode(Graph graph, int id) {
+        Node n = null;
+        
+        int nodeK = 0;
+         
+        if ( graph != null ) {
+                while ( nodeK < graph.getNodeCount() ) {
+
+                     for (Node currentNode : graph.getNodes()) {
+                          if ( currentNode.getStoreId() == id ) {
+                              n=currentNode;
+                              nodeK=graph.getNodeCount();
+                          }
+                      }
+
+                    ++nodeK;
+                }
+        }
+        return n;
+    }
+    
+      public Node3D searchNode3D(int id) {

```

Comment:
Agree. However i am using now generic data structures as suggested and i have optimized some code loops. Using a hashmap will significativelly change the tested code. I kindly request permission to keep as it is because the solution is stable and the code is clean.